### PR TITLE
File expenses

### PR DESCRIPTION
**Pull request by Linas490**
Working file reading. 

- Only accepts .txt format
- Expenses are separated by new line
- Expense data is separated by ','
- Expenses should be written as follows: Money, Comment, CategoryId, Year, Month, Day   

Note: after adding expenses from file the current list unfilters and unsorts itself

### CODE REVIEW

File: ExpensesTracker/Client/Pages/MonthlyExps.razor [hunk 1]
```diff
@@ -95,4 +112,66 @@
...
+
+                    foreach (string line in lines)
+                    {
+                        if (line.Length >= 6)
+                        {
+                            try
+                            {
+                                List<string> list = new List<string>();
+                                list = line.Split(',').ToList();
+                                MonthlyExp expense = new MonthlyExp { Money = Double.Parse(list.ElementAt(0)), Comment = list.ElementAt(1), CategoryId = Int32.Parse(list.ElementAt(2)), Year = Int32.Parse(list.ElementAt(3)), Month = Int32.Parse(list.ElementAt(4)), Day = Int32.Parse(list.ElementAt(5)) };
```
Comment (JokubasBu) [inline ExpensesTracker/Client/Pages/MonthlyExps.razor]: Press enter, please.

File: ExpensesTracker/Client/Pages/MonthlyExps.razor [hunk 2]
```diff
@@ -95,4 +112,66 @@
         await MonthlyExpService.DeleteExpense(id);
         NavigationManager.NavigateTo("/monthlyexp");
     }
+
+    private void OnInputFileChange(InputFileChangeEventArgs e)
+    {
+        selectedFiles = e.GetMultipleFiles();
+        message = "File selected";
```
Comment (JokubasBu) [inline ExpensesTracker/Client/Pages/MonthlyExps.razor]: Not sure if, message is needed. There seems to be some build in feature near choose file button that shows chosen files. Either disable that or remove this I think.

**Review by JokubasBu — CHANGES_REQUESTED** [review]
Seems to function correctly, just needs a few slight changes before merging. Also, this could be refactored into service, but this is okay too probably.

File: ExpensesTracker/Client/Pages/MonthlyExps.razor (hunk 2 above)
Comment (Linas490) [reply ExpensesTracker/Client/Pages/MonthlyExps.razor]: The message is there to not only inform user about selected and uploaded files, but also to tell about errors from wrong data input/format.

Comment (JokubasBu) [reply ExpensesTracker/Client/Pages/MonthlyExps.razor]: Then you could just make it appear only when an error occurs.

**Review by JokubasBu — APPROVED** [review]

**Pull request merged**